KIPs/kip-149.md

```diff
@@ -244,6 +244,80 @@ func ReadActiveAddressFromRegistry(backend bind.ContractCaller, name string, num
 }
 ```
 
+#### JSON-RPC APIs
+
+The following JSON-RPC methods for the Klaytn node should be added to provide the records of registered system contracts.
+
+1. `klay_getActiveAddressFromRegistry`
+
+    - Parameters:
+        - `name`: the name of the system contract in string.
+        - `number`: (optional) integer or hexadecimal block number, or the string "pending" or "latest".
+    - Description: Returns the active address of the system contract registered as `name` if exists.
+    - Return: The address of the active system contract.
+    - Example
+        ```json
+        // Request
+        curl -X POST -H "Content-Type: application/json" --data '{"jsonrpc":"2.0", "method":"klay_getActiveAddressFromRegistry", "params":["KIP113", "latest"],"id":1}' http://localhost:8551
+        // Response
+        {
+            "jsonrpc":"2.0",
+            "id":1,
+            "result": "0x0000000000000000000000000000000000000402"
+        }
+
+        // Request - no active system contract
+        curl -X POST -H "Content-Type: application/json" --data '{"jsonrpc":"2.0", "method":"klay_getActiveAddressFromRegistry", "params":["KIP114", "latest"],"id":1}' http://localhost:8551
+        // Response
+        {
+            "jsonrpc":"2.0",
+            "id":1,
+            "error":{
+                "code":-32000,
+                "message":"no active address for KIP114"
+            }
+        }
+        ```
+2. `klay_getAllRecordsFromRegistry`
+
+    - Parameters:
+        - `name`: the name of the system contract in string.
```

Comment:
1. Can we put multiple names on this param? 
2. Can we have some examples of the names? it is only about KIP-XXX names only? Is there any other name types?

Reply:
1. Do you suggest using `names = ["KIPxxx", "KIPyyy, "KIPzzz"]` for this api? It's possible but I'm not sure if that's better. If do so, we might need to change api name `ex) getAllRecordsByNamesFromRegistry`. 1 to N looks more intuitive I think unless we have specific needs.
2. It can be any registered system contracts. I choose KIP113 since it's the only system contract we have for now.

Reply:
1. No, I'm not suggesting. I am just curious how it works and what kinds of possibility we have. :) So, if you think multiple names are unnecessary, please leave it as it is.
2. I'm wondering what kinds of forms we can put into. So, in general, the name format would be KIPxxx. Is this right?

Reply:
1. Aha, I understood. Thanks.
2. Yes, I expect `KIPxxx` will be most common case since system contracts are mostly derived from KIP.